partcad/src/partcad/utils_ocp.py

```
@@ -0,0 +1,106 @@
+    from OCP.TopoDS import TopoDS
+    from OCP.TopExp import TopExp
+
+    face_map = TopTools_IndexedMapOfShape()
+    TopExp.MapShapes_s(shape, TopAbs_FACE, face_map)
+
+    for i in range(1, extent_or_size(face_map) + 1):
+        yield TopoDS.Face_s(face_map.FindKey(i))
+
+
+def tessellate(
+    shape, tolerance: float = 0.1, angularTolerance: float = 0.1
```

Ruff (0.8.2)

Missing type annotation for function argument `shape`

(ANN001)